### PR TITLE
beacon example

### PR DESCRIPTION
**Pull request by benjaminbollen**
[Review on Reviewable](https://reviewable.io/reviews/dirvine/crust/23)

### CODE REVIEW

File: examples/beacon_server.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+fn main() {
+  spawn(move || {
+    let normal_socket = match UdpSocket::bind("::0:0") {
+      Ok(socket) => socket,
+      Err(e) => panic!("Couldn't bind socket: {}", e)
+    };
+
+    match normal_socket.local_addr() {
+      Ok(local_addr) => crust::beacon::listen_for_broadcast(local_addr),
```
Comment (benjaminbollen) [inline examples/beacon_server.rs]: I don't seem to reach this point, neither do I get panics;  are panics in a spawned off thread invisible on terminal?

**Comment by benjaminbollen** [comment]
works on one machine.  will test now on several machines, cross platform with multiple nodes

**Comment by rossmuir** [comment]
As per yesterdays discussion - should `#deny(missing_docs)` be added to this PR?

File: src/beacon.rs [hunk 2]
```diff
@@ -16,6 +16,8 @@
 // See the Licences for the specific language governing permissions and limitations relating to
 // use of the MaidSafe Software.
 
+#![deny(missing_docs)]
```
Comment (rossmuir) [inline src/beacon.rs]: Sorry @benjaminbollen my bad - can we switch this on in the `lib.rs` file instead

File: examples/beacon_server.rs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+//
+// See the Licences for the specific language governing permissions and limitations relating to
+// use of the MaidSafe Software.
+
+extern crate crust;
+
+use std::net::UdpSocket;
+
+fn main() {
+  let normal_socket = match UdpSocket::bind("::0:0") {
```
Comment (chandraprakash) [inline examples/beacon_server.rs]: More realistic example will be to use supported protocol. So using tcp here will be handy.  
Later on bootstrap integration can also be used here too.

Comment (dirvine) [inline examples/beacon_server.rs]: Agreed and IPv4 please! IPv6 can be later iteration

Comment (benjaminbollen) [inline examples/beacon_server.rs]: Alright, seen the comment.  I'll address it!

File: src/beacon.rs (hunk 2 above)
Comment (benjaminbollen) [inline src/beacon.rs]: I will make a new pull request to document instead

**Comment by chandraprakash** [comment]
good to go ..

**Pull request merged**